File: src/main/java/toker/warbandscripts/panel/entity/NoticeBoardEntry.java
package toker.warbandscripts.panel.entity;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "notice_board_entry")
public class NoticeBoardEntry {
    private Integer id;
    private String entryText;
    private Integer entryNo;
    private NoticeBoard noticeBoardByBoardId;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    @Basic
    @Column(name = "entry_text", nullable = false, length = 256)
    public String getEntryText() {
        return entryText;
    }

    public void setEntryText(String entryText) {
        this.entryText = entryText;
    }

    @Basic
    @Column(name = "entry_no", nullable = false)
    public Integer getEntryNo() {
        return entryNo;
    }

    public void setEntryNo(Integer entryNo) {
        this.entryNo = entryNo;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        NoticeBoardEntry that = (NoticeBoardEntry) o;
        return Objects.equals(id, that.id) &&
                Objects.equals(entryText, that.entryText) &&
                Objects.equals(entryNo, that.entryNo);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, entryText, entryNo);
    }

    @ManyToOne
    @JoinColumn(name = "board_id", referencedColumnName = "id", nullable = false)
    public NoticeBoard getNoticeBoardByBoardId() {
        return noticeBoardByBoardId;
    }

    public void setNoticeBoardByBoardId(NoticeBoard noticeBoardByBoardId) {
        this.noticeBoardByBoardId = noticeBoardByBoardId;
    }
}

File: src/main/java/toker/panel/entity/NoticeBoardEntry.kt
package toker.panel.entity

import javax.persistence.*

@Entity
@Table(name = "notice_board_entry")
class NoticeBoardEntry(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Column(name = "id", nullable = false)
        var id: Int? = null,

        @Column(name = "entry_text", nullable = false, length = 256)
        var entryText: String? = null,

        @Column(name = "entry_no", nullable = false)
        var entryNo: Int? = null,

        @ManyToOne
        @JoinColumns(JoinColumn(name = "board_index", referencedColumnName = "index", nullable = false), JoinColumn(name = "board_server_id", referencedColumnName = "server_id", nullable = false))
        var board: NoticeBoard? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as NoticeBoardEntry
        if (id != other.id) return false
        return true
    }

    override fun hashCode(): Int {
        return id ?: 0
    }
}